File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int result = 0;

        for (List<String> item : items) {
            if (ruleKey.equals("type") && item.get(0).equals(ruleValue)) result++;
            if (ruleKey.equals("color") && item.get(1).equals(ruleValue)) result++;
            if (ruleKey.equals("name") && item.get(2).equals(ruleValue)) result++;
        }
        
        return result;
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
	result := 0

	for i := 0; i < len(items); i++ {
		itemArr := items[i]
		if ruleKey == "type" && ruleValue == itemArr[0] {
			result += 1
		} else if ruleKey == "color" && ruleValue == itemArr[1] {
			result += 1
		} else if ruleKey == "name" && ruleValue == itemArr[2] {
			result += 1
		}
	}

	return result
}
